deliver/2017-06-12-jtemporal-anaschwendler-taxi-uber-bsb-analysis.py

```diff
@@ -0,0 +1,148 @@
+
+# coding: utf-8
+
+# # Taxi and Uber expenses in Brasília analysis
+
+# In[1]:
+
+import pandas as pd
+import numpy as np
+
+from serenata_toolbox.datasets import fetch
+
+
+fetch('2017-05-25-reimbursements.xz', '../data')
+fetch('2017-05-21-companies-no-geolocation.xz', '../data') # we won't need geolocation for this analysis
+
+
+# In[2]:
+
+reimbursements = pd.read_csv('../data/2017-05-25-reimbursements.xz',
+                             dtype={'applicant_id': np.str,
+                                    'cnpj_cpf': np.str,
+                                    'congressperson_id': np.str,
+                                    'subquota_number': np.str},
+                             low_memory=False)
+
+
+# In[3]:
+
+companies = pd.read_csv('../data/2017-05-21-companies-no-geolocation.xz', low_memory=False)
+companies['cnpj'] = companies['cnpj'].str.replace(r'\D', '')
+
+
+# For this analysis Tony wanted the `Taxi, toll and parking` reimbursements from June 2014 to the present day. Let's filter that.
+
+# In[4]:
+
+reimbursements = reimbursements[reimbursements['subquota_description'] == 'Taxi, toll and parking']
+reimbursements2014 = reimbursements[reimbursements['year'] == 2014]
+reimbursements2014 = reimbursements2014[reimbursements2014['month'] >= 6]
+reimbursementsall = reimbursements[reimbursements['year'] >= 2015]
+reimbursements = reimbursements2014.append(reimbursementsall)
+reimbursements.head()
+
+
+# In[5]:
+
+reimbursements.shape
+
+
+# We want to find the top 500 companies we have most expenses for
+
+# In[6]:
+
+aggregation = reimbursements.groupby('cnpj_cpf')['total_net_value'].agg(np.sum).rename('sum').reset_index()
+top500 = aggregation.sort_values('sum', ascending=False).head(500)
+top500 = top500.reset_index(drop=True)
+top500.head()
+
+
+# In[7]:
+
+dataset = pd.merge(top500, companies, how='left', left_on='cnpj_cpf', right_on='cnpj')
+dataset = dataset.drop('cnpj', axis=1)
+dataset.head()
+
+
+# In[8]:
+
+dataset.shape
+
+
+# ## Further investigation
+# 
+# With that table we selected a few companies and apps for futher investigation:
```

Comment:
This notebook will be used as basis for an article, fell it exists more for an "How did you get this data?" kind of question once the article is out than an exploratory analysis which requires proper explanations.